pkg/agent/proxy/proxier.go

```diff
@@ -394,7 +405,10 @@ func (p *proxier) removeStaleConntrackEntries(svcPortName k8sproxy.ServicePortNa
 		svcPortChanged = true
 	} else {
 		// If the port of the Service is not changed, delete the conntrack entries related to the stale Service IPs and
-		// the Service port. These stale Service IPs could be from external IPs and loadBalancer IPs.
+		// the Service port. These stale Service IPs could be clusterIP, externalIPs or loadBalancerIPs.
+		if pClusterIP != clusterIP {
```

Comment:
I remember ClusterIP can't change once set, it's indeed possible to add/delete a clusterIP of secondary IPFamily.
Then one of `pClusterIP` and `clusterIP` will be nil in this case, which could cause some issues with the current code, I suppose?

Reply:
> I remember ClusterIP can't change once set, it's indeed possible to add/delete a clusterIP of secondary IPFamily.

You are right. I have verified that. I have tested it with the following yamls. The clusterIP of the secondary IPFamlily can be added or deleted.

The dual stack Service:
```yaml
apiVersion: v1
kind: Service
metadata:
  name: clusterip
spec:
  ipFamilyPolicy: PreferDualStack
  ipFamilies:
    - IPv6
    - IPv4
  clusterIP: 2010:96::3052
  selector:
    app: nginx
  ports:
    - port: 80
      protocol: TCP
      targetPort: 80
```

The single stack Service:
```yaml
apiVersion: v1
kind: Service
metadata:
  name: clusterip
spec:
  ipFamilyPolicy: SingleStack
  ipFamilies:
    - IPv6
  clusterIP: 2010:96::3052
  selector:
    app: nginx
  ports:
    - port: 80
      protocol: TCP
      targetPort: 80
```

However, I have verified that when changing a dual stack Service to single stack Service, the `ServiceInfo` of the secondary IPFamilly, consumed in `syncProxyRules`, is marked as "deleted" since there is no valid clusterIP of the secondary IPFamiliy. The code is here:

https://github.com/antrea-io/antrea/blob/602bfa49844233ba7b95a9e667bdcc6f0efc1bbc/third_party/proxy/service.go#L447

As a result, `pClusterIP` and `clusterIP` should not be nil.

Reply:
I added a new unit test for the case that dual stack Service to single stack Service in a new commit.

Reply:
then this new check is unnecessary given clusterIP is unchangable?

Reply:
The clusterIP is unchangable, but we have similar code in existing code https://github.com/antrea-io/antrea/blob/d895124624ac54a6a9c50d24fb810954771a9666/pkg/agent/proxy/proxier.go#L485.

For my understanding (not fully sure), the changes of a Service with the same name could be accumulated. For example, a Service could be added, deleted and added again, if the first and the second events are not synced in time, and the sync is done with the third event, it is likely that the `clusterIP` is changed.

Reply:
You are right.